python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -905,10 +904,24 @@ def get_event_records(
                         "Could not resolve event record as EventLogEntry for id `%s`.", row_id
                     )
                     continue
-                else:
-                    event_records.append(
-                        EventLogRecord(storage_id=row_id, event_log_entry=event_record)
-                    )
+
+                if event_records_filter.tags and not self.has_table(AssetEventTagsTable.name):
+                    # If we can't filter tags via the tags table, filter the returned records
```

Comment:
Can we add comments to this function highlighting that any tags that do not exist on the materialization event object will not be returned?

I think in `add_asset_event_tags` it would also be helpful to comment that adding tags after the materialization already exists should be done with caution (as this `get_event_records` call won't find tags outside of the materialization event if the asset event tags table doesn't exist)

Reply:
Would it ever be possible to end up in that situation? It looks like `add_asset_event_tags` will fail when there's no asset event tags table.

Reply:
ah gotcha, thought that it would silently exit in that case. this is looking fine to me then!